### ISSUE TITLE
[🌟] video compression on selecting from device storeage

### ISSUE THREAD

**Issue by siddharth-kt** [react-native-image-picker/react-native-image-picker#1831]
### Description

This library is working excellent, but the only thing missing is video compression when file is selected from device storage (I noticed it on android), It is compressing image (both clicked[launchCamera()] or from device storage[launchImageLibrary()]) + video (only recorded from launchCamera() and not from device storage), it will be great to have this feature in this library.

**Comment by CristianAThompson** [comment]
I too noticed this is occurring on our iOS version as well, it works well from the camera, but does not compress video when selecting from the imagepicker

**Comment by pierroo** [comment]
Did you face any solution to this issue since @siddharth-kt ?
That would be very helpful :)

**Comment by siddharth-kt** [comment]
@pierroo i am using this lib for compression : https://github.com/shahen94/react-native-video-processing

**Comment by pierroo** [comment]
I was afraid you would answer that @siddharth-kt ; this library being 103Mb, is adding 50Mb to the final bundle size.
Did you not face this issue when releasing your app?
Your feedback is appreciated :)

**Comment by siddharth-kt** [comment]
yeah size is a problem but can't do much. I googled for how to make react native aab size small (helped a bit) but not much.

**Comment by pierroo** [comment]
Okay thank you for that.
Ideally react-native-image-picker could handle that natively, but unfortunately no one seems concerned about it :/
The best alternative to react-native-video-processing I found seems to be https://github.com/tanersener/ffmpeg-kit, but this one is infinitely harder to use unfortunately.
Thanks anyway!

**Comment by siddharth-kt** [comment]
@pierroo if you decided to uses https://github.com/tanersener/ffmpeg-kit then kindly share the code with me.

**Comment by pierroo** [comment]
I will, if ever I look into it further, for now I completely abandoned the idea since it is way beyond my knowledge unfortunately.

**Comment by siddharth-kt** [comment]
@pierroo ok

**Comment by sushilbansal** [comment]
you can try this package (850kb): https://www.npmjs.com/package/react-native-compressor
Considering its usage for my project.

**Comment by pierroo** [comment]
> you can try this package (850kb): https://www.npmjs.com/package/react-native-compressor Considering its usage for my project.

Thank you for your suggestion @sushilbansal , I confirm this is an awesome package for compressing photos.
Unfortunately quite a few issue with compressing videos and photos on iOS, as stated here:

- https://github.com/Shobbak/react-native-compressor/issues/109
- https://github.com/Shobbak/react-native-compressor/issues/108

Other worrying point with this package, is that clearly the owner has stopped answering all the bugs ticket for a long time now. :(

**Comment by sushilbansal** [comment]
somebody wrote an article on the exact process i followed. I found it afterwards: https://medium.com/@islamrustamov/how-to-convert-and-upload-videos-in-react-native-7c2440bcdc1b

Let me know if you face any issues with understanding the code provided in the article.

**Comment by siddharth-kt** [comment]
@sushilbansal thanks, i will ask you later if i face any issue.

**Comment by numandev1** [comment]
@sushilbansal @pierroo 🙏 🙏 sorry a long ago I was not maintaining [react-native-compressor](https://github.com/numandev1/react-native-compressor) due to office workload, hopefully, i am going to start fixing bugs on weekend basis

**Comment by viniciusaugutisps** [comment]
@numandev1 I'm using the lib [react-native-compressor](https://github.com/numandev1/react-native-compressor) and I saw that for Android the video compression is very slow, making it unfeasible to use. Being able to support that point for us, I'm grateful.

**Comment by numandev1** [comment]
@viniciusaugutisps I fixed the slow problem on android, did you check the latest version of [react-native-compressor](https://github.com/numandev1/react-native-compressor)?

**Comment by numandev1** [comment]
> Okay thank you for that. Ideally react-native-image-picker could handle that natively, but unfortunately no one seems concerned about it :/ The best alternative to react-native-video-processing I found seems to be https://github.com/tanersener/ffmpeg-kit, but this one is infinitely harder to use unfortunately. Thanks anyway!

ffmpeg increased 8 to 10MB size in apk, don't use this, I fixed almost all the issues in [react-native-compressor](https://github.com/numandev1/react-native-compressor), you can give it a try
